pkg/master/controller/crdregistration/crdregistration_controller.go

```diff
@@ -195,26 +198,25 @@ func (c *crdRegistrationController) handleVersionUpdate(groupVersion schema.Grou
 		return err
 	}
 	for _, crd := range crds {
-		if crd.Spec.Version == groupVersion.Version && crd.Spec.Group == groupVersion.Group {
-			found = true
-			break
+		if crd.Spec.Group != groupVersion.Group {
+			continue
+		}
+		for _, version := range crd.Spec.Versions {
+			if version.Name == groupVersion.Version && version.Served {
+				c.apiServiceRegistration.AddAPIServiceToSync(&apiregistration.APIService{
+					ObjectMeta: metav1.ObjectMeta{Name: apiServiceName},
+					Spec: apiregistration.APIServiceSpec{
+						Group:                groupVersion.Group,
+						Version:              groupVersion.Version,
+						GroupPriorityMinimum: 1000, // CRDs should have relatively low priority
+						VersionPriority:      100,  // CRDs should have relatively low priority
```

Comment:
Done

Reply:
thinking through this more, what happens if two CRDs define the same group versions, but in opposite order? 

For example:
* CRD A defines group=foo, versions=v1,v2
* CRD B defines group=foo, versions=v2,v1

What should the versionpriority for foo/v1 and foo/v2 be? How do we ensure the version priority for the API service is deterministic no matter which CRD happens to come first when searching?

One possibility is to keep track of the version priority calculated for a particular CRD but keep searching through all the CRDs in case any others define it with a higher priority. Highest priority is used. That keeps it deterministic, and if all CRDs for a group define the same versions in the same order, that becomes the order in discovery

Reply:
> I'd start with 100 to avoid fighting previous API servers and add 1 for every served version encountered

I also realized I'd flipped priority in my head... higher priority items come first. Given that version priority can't be less than 0, `100` seems slightly low as a starting point for multi-version CRDs.
* For single-version CRDs, we should still stick with `VersionPriority: 100` to avoid dueling with old apiservers.
* For multi-version CRDs, we could pick a high version priority (1000? 10000? how crazy do we think people will go with CRDs versions?) and *subtract* 1 for every served version encountered.

Reply:
What about just adding a `Priority` field to the `CustomResourceDefinitionVersion` type? If user does not set it, we can auto-fill it.